### ISSUE TITLE
"reused" translation warning with empty list and resist to reset

### ISSUE THREAD

**Issue by yilmazdurmaz** [WeblateOrg/weblate#9621]
### Describe the issue

https://hosted.weblate.org/translate/godot-engine/godot/tr/?q=Create+a+Tile&sort_by=-priority%2Cposition&offset=12

there was a variant of this with small first letters: https://hosted.weblate.org/translate/godot-engine/godot/tr/?q=Create+a+Tile&sort_by=-priority%2Cposition&offset=6

both had the wrong translation.

I corrected the small-lettered first, but missed the size and made it capitalized.

after I corrected the actual capitalized translation, it collided with the other should-be-small-lettered translation and gave me the "reused translation warning".

I marked it to ignore for the time, and continued working with other strings.

I later went back and corrected small-lettered translation.

And Voila! The Bug. now the capitalized version has this empty "other sources" line and it rejects me from resetting it. I press reset, save the string again, change the page and come back. 

The translation does not "resaved" and the "reused" warning is still insisting not to go away.

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar issues in this repository.

### Steps to reproduce the behavior

1. have 2 similar strings, A and B
2. translate A  then B to the same content. B should get the "reused" warning
3. press "dismiss" button first
4. the go to A and change it a bit.
5. come back to B and "source" is empty and warning wont go

### Expected behavior

reused warnings on a string must resolve by self if the other string has changed.

### Screenshots

![empty reused translation warningweblate](https://github.com/WeblateOrg/weblate/assets/1812105/e684aede-f461-4af6-9bdf-bcd704181c2a)

### How do you run Weblate?

weblate.org service

**Comment by nijel** [comment]
Bug confirmed, the check propagation does not work well for reused check.

**Comment by yilmazdurmaz** [comment]
@nijel  Hi again,

I guess your fix is alive, but it seems the problem avoided the fix because it is still there. I had to edit one of these strings with an extra "." and then reverted this change, adding 2 entries to the change history.

As I mentioned, the step is "edit string A, then string B". But this has a gotcha: Only string B has the first reused warning.

Since string A does not have this warning (it has no idea it is reused), applying "go to string A and change it" will propagate nowhere, or maybe it forgets old content and propagates with its new content. the system knows about the change because B now has an empty list.

I could not follow the code in which order it saves and propagates. for this "reused" issue, I think, it should be "first propagate the change for the old state, then check and save the new state".

On the other hand, this might come with performance issues. another way to fix this might be adding an "on-demand check" button onto the "things to check" panel: A "check again" button to trigger checks for the current string, for all checks, or for only the reused check.

**Comment by yilmazdurmaz** [comment]
how long does it take before any direct commit goes live on the "hosted.weblate.org"?

**Comment by nijel** [comment]
There is no automated deploy, so it depends.

**Comment by yilmazdurmaz** [comment]
Sorry to bear the bad news: The problem is still there.

Let me shorten my last post:
1- string B knows it is a duplicate, but string A does not know it has a duplicate.
2- thus, changing A does not send a signal to B
3- or it sends a signal but with the wrong content. it should:
  - first send a signal to all strings that may have its "old" content so they may remove their duplicate tags (I think this is the missing part)
  - then send a signal to all strings that may have its "new" content so they may add duplicate tags to themselves.

**Comment by nijel** [comment]
Yes, the problem is still there, that's why this issue is reopened.

**Comment by timothyqiu** [comment]
I still encounter this issue today.

1. String A, B, C shares the same fuzzy translation (needs edit)
2. Save translation for string A
   * Reused translation: other source strings B, C
3. Save translation B and C as different strings
    * String A still complains about reused translation with an empty list

**Comment by nijel** [comment]
Can you please share a link to such string?

**Comment by timothyqiu** [comment]
String A, B and C accordingly:
* https://hosted.weblate.org/translate/godot-engine/godot-class-reference/zh_Hans/?checksum=e187f7c8d0a6c819
* https://hosted.weblate.org/translate/godot-engine/godot-class-reference/zh_Hans/?checksum=aae1f02be89ba04b
* https://hosted.weblate.org/translate/godot-engine/godot-class-reference/zh_Hans/?checksum=ac0d459f4f5c44c5

**Comment by nijel** [comment]
This case should be fixed by bb31bcd97e

**Comment by yilmazdurmaz** [comment]
It is now also active on the "hosted".
Thanks for the fix.

**Comment by yilmazdurmaz** [comment]
@nijel , Have you revert this fix? Or the last time I thought it worked was an illusion I had?

it is back on more than one string. These are two of them
https://hosted.weblate.org/translate/godot-engine/godot-properties/tr/?q=check%3Areused&offset=1
https://hosted.weblate.org/translate/godot-engine/godot-properties/tr/?q=check%3Areused&offset=2

**Comment by nijel** [comment]
24ff8222550c05b8011a3306d706932de6fb326a should have fixed outstanding issues.
